repose-aggregator/components/filters/header-id-mapping/src/main/scala/org/openrepose/filters/headeridmapping/HeaderIdMappingFilter.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2016 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+
+package org.openrepose.filters.headeridmapping
+
+import java.net.URL
+import javax.inject.{Inject, Named}
+import javax.servlet._
+import javax.servlet.http.{HttpServletResponse, HttpServletRequest}
+
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import org.openrepose.commons.config.manager.UpdateListener
+import org.openrepose.commons.utils.http.PowerApiHeader
+import org.openrepose.commons.utils.servlet.http.HttpServletRequestWrapper
+import org.openrepose.core.filter.FilterConfigHelper
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.filters.headeridmapping.config.{HttpHeader, HeaderIdMappingConfig}
+
+import scala.collection.JavaConverters._
+
+@Named
+class HeaderIdMappingFilter @Inject()(configurationService: ConfigurationService)
+  extends Filter with UpdateListener[HeaderIdMappingConfig] with LazyLogging {
+
+  import HeaderIdMappingFilter._
+
+  private var configurationFile: String = DEFAULT_CONFIG
+  private var initialized = false
+  private var configuredHeaders: Iterable[HttpHeader] = _
```

Comment:
Additional thoughts: do we have a reason to use AtomicThings rather than volatile (or the @volatile annotation in Scala)? I suppose we could also lock with sychronized. If we're only using .get() and .set(...) on AtomicThings, though, those three options should be equivalent, I think?

Also, wouldn't making "initialized" volatile suffice? Assuming we set initialized at the end of the configurationUpdated(...) method. Although it's probably clearer to mark all of the fields which may be mutated and read by multiple threads as volatile. Does doing so hamper performance? I wouldn't think so since it's the same block of code being locked, as it were.